### PR TITLE
fix(secrets): Change s3 path for enriched secrets upload

### PR DESCRIPTION
**Pull request by Eliran-Turgeman**
**By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.**

[//]: # "
    # PR Title
    Be aware that we use the title to create changelog automatically and therefore only allow specific prefixes
    - break:    to indicate a breaking change, this supersedes any of the types
    - feat:     to indicate new features or checks
    - fix:      to indicate a bugfix or handling of edge cases of existing checks
    - docs:     to indicate an update to our documentation
    - chore:    to indicate adjustments to workflow files or dependency updates
    - platform: to indicate a change needed for the platform
    Additionally a scope is needs to be added to the prefix, which indicates the targeted framework, in doubt choose 'general'.
    #    
    Allowed prefixs:
    argo|arm|azure|bicep|bitbucket|circleci|cloudformation|dockerfile|github|gha|gitlab|helm|kubernetes|kustomize|openapi|sca|secrets|serverless|terraform|general|graph|terraform_plan
    #
    ex.
    feat(terraform): add CKV_AWS_123 to ensure that VPC Endpoint Service is configured for Manual Acceptance
"

## Description
* Change enriched secrets object s3 upload path
* Upload enriched secrets only for customers eligible for secrets verifier
**Dependent changes has to be merged before merging this PR**

*Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.*

Fixes # (issue)

### Description
*Include a description of what makes it a violation and any relevant external links.*

### Fix
*How does someone fix the issue in code and/or in runtime?*

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my feature, policy, or fix is effective and works
- [x] New and existing tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Review by marynaKK — APPROVED** [review]

File: checkov/common/bridgecrew/platform_integration.py [hunk 1]
```diff
@@ -422,8 +423,12 @@ def persist_enriched_secrets(self, enriched_secrets: list[EnrichedSecret]) -> st
...
             return None
 
-        repo_path_without_src = os.path.dirname(self.repo_path)
-        s3_path = f'{repo_path_without_src}/{checkov_results_prefix}/{CheckType.SECRETS}/secrets_to_verify.json'
+        if not bc_integration.bc_api_key or not os.getenv("CKV_VALIDATE_SECRETS"):
+            logging.debug('Skipping persistence of enriched secrets object as secrets verification is off,'
+                          ' enabled it via env var CKV_VALIDATE_SECRETS and provide an api key')
+            return None
+
+        s3_path = f'checkov/original_secrets/{uuid.uuid4()}.json'
```
Comment (gruebel) [inline checkov/common/bridgecrew/platform_integration.py:431]: let's talk about this

Comment (Eliran-Turgeman) [reply checkov/common/bridgecrew/platform_integration.py]: changed the path, example: `original_secrets/elturgeman5/elturgeman5_cli_repo/verifier/1674031834070/686fd705-c869-4ed3-9b07-528ff5397013.json`

File: checkov/common/bridgecrew/platform_integration.py [hunk 2]
```diff
@@ -422,8 +423,13 @@ def persist_enriched_secrets(self, enriched_secrets: list[EnrichedSecret]) -> st
...
             return None
 
-        repo_path_without_src = os.path.dirname(self.repo_path)
-        s3_path = f'{repo_path_without_src}/{checkov_results_prefix}/{CheckType.SECRETS}/secrets_to_verify.json'
+        if not bc_integration.bc_api_key or not os.getenv("CKV_VALIDATE_SECRETS"):
+            logging.debug('Skipping persistence of enriched secrets object as secrets verification is off,'
+                          ' enabled it via env var CKV_VALIDATE_SECRETS and provide an api key')
+            return None
+
+        base_path = self.repo_path.replace('checkov/', 'original_secrets/').replace('/src', '')
```
Comment (gruebel) [inline checkov/common/bridgecrew/platform_integration.py:431]:
```suggestion
        base_path = self.repo_path.replace('checkov/', 'original_secrets/').replace('/src', '')
```
can you make sure only to replace `checkov/` at the beginning and only remove `/src` at the end. probably easiest to do is with `re.sub()`

Comment (Eliran-Turgeman) [reply checkov/common/bridgecrew/platform_integration.py]: done :)

File: .flake8 [hunk 3]
```diff
@@ -3,7 +3,7 @@
 [flake8]
 max-line-length = 120
 # E203,E501 don't work with black together
-ignore = E203,E501,E731,W293,W503,W504,DUO107,DUO104,DUO130,DUO109,DUO116,B028,B950,TC001,TC003,TC006
+ignore = E203,E501,E731,W293,W503,W504,DUO107,DUO104,DUO130,DUO109,DUO116,B028,B950,TC001,TC003,TC006,B907
```
[the remark this answers is not in the document]
Comment (gruebel) [reply .flake8:6]: there is a chance you will need to remove this change, when we merge #4274 tomorrow

**Review by gruebel — APPROVED** [review]

File: checkov/common/bridgecrew/platform_integration.py [hunk 4]
```diff
@@ -422,9 +423,24 @@ def persist_enriched_secrets(self, enriched_secrets: list[EnrichedSecret]) -> st
...
 
-        repo_path_without_src = os.path.dirname(self.repo_path)
-        s3_path = f'{repo_path_without_src}/{checkov_results_prefix}/{CheckType.SECRETS}/secrets_to_verify.json'
-        _put_json_object(self.s3_client, enriched_secrets, self.bucket, s3_path)
+        if not bc_integration.bc_api_key or not os.getenv("CKV_VALIDATE_SECRETS"):
+            logging.debug('Skipping persistence of enriched secrets object as secrets verification is off,'
+                          ' enabled it via env var CKV_VALIDATE_SECRETS and provide an api key')
+            return None
+
+        base_path = re.sub(r'checkov/(.*?)/src', r'original_secrets/\1', self.repo_path)
```
Comment (gruebel) [inline checkov/common/bridgecrew/platform_integration.py:431]: can you just move it `r'checkov/(.*?)/src'` up to global with a `re.compile()` then linters can also pick it up better

**Review by gruebel — APPROVED** [review]

**Pull request merged**